packages/types/src/auth.ts

```diff
@@ -0,0 +1,39 @@
+import { Provider } from "./util";
+
+export interface Identity {
+  /**
+   * A {Date} when the identity or credential will no longer be accepted.
+   */
+  readonly expiration?: Date;
+}
+
+/**
+ * An object representing temporary or permanent AWS credentials.
+ */
+export interface Credentials extends Identity {
+  /**
+   * AWS access key ID
+   */
+  readonly accessKeyId: string;
+
+  /**
+   * AWS secret access key
+   */
+  readonly secretAccessKey: string;
+
+  /**
+   * A security or session token to use with these credentials. Usually
+   * present for temporary credentials.
+   */
+  readonly sessionToken?: string;
+}
+
+export type CredentialProvider = Provider<Credentials>;
+
+/**
+ * Authentication schemes represent a way that the service will authenticate the customer’s identity.
+ */
+export interface AuthScheme {
```

Comment:
Where is `AuthScheme` used?

Reply:
it will be used in updates to existing packages in the next PR, it can be seen in https://github.com/aws/aws-sdk-js-v3/pull/3927

I wanted to update types with the new code in this PR instead of in multiple PRs, so as not to miss anything copying over from the original endpoints-20 branch.